requests/models.py

```diff
@@ -796,7 +800,23 @@ def iter_lines(self, chunk_size=ITER_CHUNK_SIZE, decode_unicode=None, delimiter=
             if delimiter:
                 lines = chunk.split(delimiter)
             else:
+                # Python splitlines() supports the universal newline (PEP 278).
+                # That means, '\r', '\n', and '\r\n' are all treated as end of
+                # line. If the last chunk ends with '\r', and the current chunk
+                # starts with '\n', they should be merged and treated as only
+                # *one* new line separator '\r\n' by splitlines().
+                # This rule only applies when splitlines() is used.
+                if last_chunk_ends_with_cr and chunk.startswith(line_feed):
+                    # The last chunk ends with '\r', so the '\n' at chunk[0]
+                    # is just the second half of a '\r\n' pair rather than a
+                    # new line break. Just skip it.
+                    chunk = chunk[1:]
+                    # it's possible that after stripping the '\n' then chunk becomes empty
+                    if not chunk:
+                        continue
                 lines = chunk.splitlines()
+                # check if the current chunk ends with '\r'
+                last_chunk_ends_with_cr = chunk.endswith(carriage_return)
```

Comment:
@Lukasa Nice catch! Sorry that I overlooked such a simple check. Just add a simple fix along with a test case for it. BTW, I did not hoist the flag per your request. It will look somewhat like this:
```
                # a temp variable is needed here and I cannot find a good variable naming that is not confusing
                old_last_chunk_ends_with_cr = last_chunk_ends_with_cr
                last_chunk_ends_with_cr = chunk.endswith(carriage_return)
                if old_last_chunk_ends_with_cr and chunk.startswith(line_feed):
                    chunk = chunk[1:]
                    if not chunk:
                        last_chunk_ends_with_cr = False
                        continue
                lines = chunk.splitlines()
```
Also this yields an unnecessary method call endwiths('\r') since when chun[1:] is empty we already know the result if False. So I prefer just setting the flag to False though it looks like code duplication at first glance.